.gitlab/ci/.gitlab-ci.global.yml

```diff
@@ -210,6 +210,31 @@
   - set | grep -E "^CI_.*=|^GITLAB.*=" | sort
   - section_end "Gitlab CI Build Parameters"
 
+.checkout-upload-commit-content-nightly: &checkout-upload-commit-content-nightly
+  - section_start "Checkout upload commit content nightly"
+  - | 
+    if [[ -n "${NIGHTLY}" && "${CI_COMMIT_BRANCH}" == "install_packs_nightly_bugs" ]]; then
+      echo "content nightly: checking out to the last successful upload commit $LAST_UPLOAD_COMMIT for the Packs folder"
+      git status
+      git reset $LAST_UPLOAD_COMMIT -- ./Packs
+      git checkout -- ./Packs
```

Comment:
why not `git checkout $LAST_UPLOAD_COMMIT -- ./Packs` ? whats reset doing here

Reply:
see the last comment here: https://stackoverflow.com/questions/6119036/how-to-revert-a-folder-to-a-particular-commit-by-creating-a-patch

Reply:
You can use git reset to reset the index which will also include removing files that were added in more recent commits (git checkout on it's own doesn't do this):

Reply:
lets add comments explaining each line and why its neccessary

Reply:
done